contracts/infra/teams/ITeamsManager.sol

```diff
@@ -0,0 +1,112 @@
+     *                by an admin or owner.
+     */
+    enum Status {
+        OWNER,
+        ADMIN,
+        MEMBER
+    }
+
+    /**
+     *  @notice The parameters of an invitation to join a team.
+     *
+     *  @param member The address of the member being invited.
+     *  @param status The status to grant to the invited member.
+     *  @param deadline The deadline by which the invitation must be accepted.
+     */
+    struct Invitation {
+        // 1 storage slot
+        uint32 teamId; //  ─┐ 4 bytes
+        address member; //  | 20 bytes
+        Status status; //   | 1 bytes
+        uint56 deadline; //─┘ 7 bytes
+    }
+
+    /*///////////////////////////////////////////////////////////////
+                            External functions
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     *  @notice Create a team. Mints an ERC-1155 token representing the team.
+     *
+     *  @param owner The owner of the team.
+     *  @param metadataURI The NFT metadata URI of the team.
+     */
+    function createTeam(address owner, string calldata metadataURI) external returns (uint256 teamId);
+
+    /**
+     *  @notice Transfer a team's ownership to another address. The caller must be the owner of the team.
+     *
+     *  @param invitation The invitation to transfer ownership. The specified status must be OWNER.
+     */
+    function transferTeamOwnership(Invitation calldata invitation) external;
+
+    /**
+     *  @notice Invite a member to a team. The caller must have appropriate team status based on the status of the member being invited.
+     *
+     *  @param invitation The invitation to join the team.
+     */
+    function inviteMember(Invitation calldata invitation) external;
+
+    /**
+     *  @notice Accept an invitation to join a team.
+     *
+     *  @param teamId The ID of the team.
+     */
+    function acceptInvite(uint256 teamId) external;
+
+    /**
+     *  @notice Remove a team member. The caller must have appropriate team status based on the status of the member being removed.
+     *
+     *  @param teamId The ID of the team.
+     *  @param member The address of the member to remove.
+     */
+    function removeMember(uint256 teamId, address member) external;
+
+    /**
+     *  @notice Leave a team. The caller is removed from the team.
+     *
+     *  @param teamId The ID of the team.
+     */
+    function leaveTeam(uint256 teamId) external;
+
+    /**
+     *  @notice Deletes a team. The caller must be the owner of the team.
+     *
+     *  @param teamId The ID of the team.
+     */
+    function deleteTeam(uint256 teamId) external;
+
+    /*///////////////////////////////////////////////////////////////
+                            View functions
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice Returns the owner of a team.
+    function teamOwner(uint256 teamId) external view returns (address);
```

Comment:
not clear how this will return a single address, this is need to be enforced somehow in the implementation to return a single owner by revoking the OWNER status from the sender of which happened when the invited new OWNER accepts. This also brings the question about whether OWNER should be just another Status value like ADMIN and MEMBER or should it be managed separately so there will be a mapping between teamId and owner address

Reply:
>not clear how this will return a single address, this is need to be enforced somehow in the implementation to return a single owner

For the Team spec, each team will only ever have one owner at ay given time (it can have multiple admins and members).